### PR TITLE
add go HEAD benchmarks

### PR DESCRIPTION
**Pull request by tamird**

### CODE REVIEW

**Comment by bdarnell** [comment]
Nice. LGTM

File: README.md [hunk 1]
```diff
@@ -23,4 +25,17 @@ Run with `go test -benchmem -benchtime 5s -bench .`
 | ProtoHTTP1_1K  |  20000 |  383342 ns/op |  5.34 MB/s |  55044 B/op | 416 allocs/op |
 | ProtoHTTP1_64K |   3000 | 1932455 ns/op | 67.83 MB/s | 827150 B/op | 377 allocs/op |
 
+## go version devel +59129c6 Fri Sep 25 22:25:52 2015 +0000 darwin/amd64
```
Comment (bdarnell) [inline README.md]: Add a comment that the difference comes from the use of hardware acceleration for AES-GCM.

Comment (tamird) [inline README.md]: done

File: README.md [hunk 2]
```diff
@@ -23,4 +25,20 @@ Run with `go test -benchmem -benchtime 5s -bench .`
...
+
+These are faster than go1.5.1 thanks to cloudflare's amd64 assembly implementation of
```
Comment (petermattis) [inline README.md]: That's quite a nice speedup.

Comment (spencerkimball) [inline README.md]: :+1:

File: README.md [hunk 3]
```diff
@@ -23,4 +25,20 @@ Run with `go test -benchmem -benchtime 5s -bench .`
...
+AES-GCM: https://github.com/golang/go/commit/efeeee3.
+
+| Bench          | Iters  | Speed         | Throughput  | Allocations | Allocations    |
+|----------------|-------:|--------------:|------------:|------------:|---------------:|
+| GRPC_1K        | 100000 |   70215 ns/op |  29.17 MB/s |  16658 B/op |  114 allocs/op |
+| GRPC_64K       |  10000 |  768996 ns/op | 170.45 MB/s | 714502 B/op |  357 allocs/op |
+| GobRPC_1K      | 200000 |   39776 ns/op |  51.49 MB/s |   4744 B/op |   33 allocs/op |
+| GobRPC_64K     |  20000 |  334447 ns/op | 391.91 MB/s | 263858 B/op |   97 allocs/op |
```
Comment (spencerkimball) [inline README.md]: The efficiency of gob is really quite an impressive achievement.

File: README.md [hunk 4]
```diff
@@ -23,4 +25,20 @@ Run with `go test -benchmem -benchtime 5s -bench .`
 | ProtoHTTP1_1K  |  20000 |  383342 ns/op |  5.34 MB/s |  55044 B/op | 416 allocs/op |
 | ProtoHTTP1_64K |   3000 | 1932455 ns/op | 67.83 MB/s | 827150 B/op | 377 allocs/op |
 
+## go version devel +59129c6 Fri Sep 25 22:25:52 2015 +0000 darwin/amd64
+
+These are faster than go1.5.1 thanks to cloudflare's amd64 assembly implementation of
+AES-GCM: https://github.com/golang/go/commit/efeeee3.
+
+| Bench          | Iters  | Speed         | Throughput  | Allocations | Allocations    |
```
Comment (spencerkimball) [inline README.md]: @tamird do we have any numbers for no SSL?

Comment (tamird) [inline README.md]: an old version of this had unencrypted numbers; HTTP2 is TLS-only, so I had to even the playing field but changing everything to TLS. Old numbers: https://github.com/cockroachdb/rpc-bench/tree/d2235332f4f593f6c98653f1574855d38bb90a8f

**Pull request merged**